src/communication/types.hpp

```diff
@@ -0,0 +1,123 @@
+#ifndef COMMUNICATION_TYPES_HPP
+#define COMMUNICATION_TYPES_HPP
+
+#include <cstdint>
+#include <memory>
+#include <vector>
+
+#include <aos/common/tools/error.hpp>
+
+#include "iamclient/types.hpp"
+
+namespace aos::mp::communication {
+
+class CommChannelItf {
+public:
+    ~CommChannelItf() = default;
+
+    virtual aos::Error Connect() = 0;
+
+    /**
+     * Reads message.
+     *
+     * @param message Message.
+     * @return aos::Error.
+     */
+    virtual aos::Error Read(std::vector<uint8_t>& message) = 0;
```

Comment:
Looks like std::vector<uint8_t> is widely used in types.
Consider defining an alias
for example,
`using bytes = std::vector<uint8_t>;`

Reply:
Yes, you're correct. However, the main issue is that the type std::vector<uint8_t> is used across different modules, and there's no common location to add this alias. Creating a utility solely for this alias seems redundant to me, so I've decided to leave it as is.